src/Symfony/Bridge/Twig/AppVariable.php

```diff
@@ -89,6 +91,8 @@ public function getUser()
         if (null === $this->tokenStorage) {
             if (null === $this->container) {
                 throw new \RuntimeException('The "app.user" variable is not available.');
+            } elseif (!$this->container->has('security.context')) {
+                return;
```

Comment:
But I'm indeed wondering why we should return a `RuntimeException` in the above case where the container isn't defined. When this particular case might happen ? Does it make sense to throw this exception instead of returning `null`?

Obviously, if this PR is merged as it is, we have some kind of inconsistency. It feels odd, and the same might be applicable to `request` and `session` globals...